health/health_log.c

```diff
@@ -364,6 +370,20 @@ static inline ssize_t health_alarm_log_read(RRDHOST *host, FILE *fp, const char
 
             ae->last_repeat = last_repeat;
 
+            if (likely(entries > 28)) {
+                freez(ae->class);
+                ae->class       = strdupz(pointers[28]);
+                if(!*ae->class) { freez(ae->class); ae->class = NULL; }
```

Comment:
You do not need to do this test here when some `NAMEz` function is called, because when it is not possible to allocate memory, Netdata will shutdown itself, please, take a look [here](https://github.com/netdata/netdata/blob/7b913eaf8b1f94af1f99b3acfa90ae36e7c47706/libnetdata/libnetdata.c#L165-L169), [here](https://github.com/netdata/netdata/blob/7b913eaf8b1f94af1f99b3acfa90ae36e7c47706/libnetdata/log/log.h#L95), and  [here](https://github.com/netdata/netdata/blob/7b913eaf8b1f94af1f99b3acfa90ae36e7c47706/libnetdata/log/log.c#L795-L840).
Any doubt, please, let me know.

Reply:
Thanks Thiago. To be honest, I followed the pattern of reading the other values (i.e. `ae-exec`, `ae->recipient`, etc), a few lines above. I think the check is on the value itself, i.e. if the strdupz returns something but its empty, set the pointer to null... ?

Reply:
Hello @MrZammler ,

The `strdupz` won't return `NULL`, because the `strdupz()` will call  `fatal()` that reports the problem and finally calls  `netdata_cleanup_and_exit(1);` . 

Our `NAMEz` functions differ of the traditional standard library, because it exits when memory is not allocated, while standard library returns NULL.

Reply:
Its `!*ae->class` (notice the *) -- checking for a zero length string